r2-core/src/main/java/com/linkedin/r2/transport/http/client/AsyncPoolImpl.java

```diff
@@ -840,15 +850,15 @@ public long getTime()
 
     private WaiterTimeoutCallback(final Callback<T> callback)
     {
-      _timeout = new SingleTimeout<>(_timeoutExecutor, _waiterTimeout, TimeUnit.MILLISECONDS, callback, () -> {
+      _timeout = new SingleTimeout<>(_timeoutExecutor, _waiterTimeout, TimeUnit.MILLISECONDS, callback, (callback1) -> {
 
         synchronized (_lock)
         {
           _waiters.remove(this);
           _statsTracker.incrementWaiterTimedOut();
         }
         LOG.debug("{}: failing waiter due to waiter timeout", _poolName);
-        callback.onError(
+        callback1.onError(
```

Comment:
I would rather not put a number in the callback variable, something like: callbackIfTimeout